src/marshmallow_union/__init__.py

```diff
@@ -59,9 +59,16 @@ def _serialize(self, value: t.Any, attr: str, obj: str, **kwargs):
         for candidate_field in fields:
 
             try:
-                return candidate_field.serialize(
-                    attr, obj, error_store=error_store, **kwargs
-                )
+                try:
+                    return candidate_field.serialize(
+                        attr, obj, error_store=error_store, **kwargs
+                    )
+                except TypeError:
+                    if attr is obj is None:  # is a mapping value
+                        return candidate_field._serialize(
```

Comment:
This fails the pylint check. Add `pylint: disable=protected-access` on that line.